kroxylicious/src/test/java/io/kroxylicious/proxy/micrometer/StandardBindersHookTest.java

```diff
@@ -25,29 +25,33 @@ public void testNullHookConfigThrows() {
 
     @Test
     public void testNullBinderNames() {
-        StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(null));
-        MeterRegistry registry = whenRegistryConfiguredWith(hook);
-        thenNoMetersRegistered(registry);
+        try (StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(null))) {
+            MeterRegistry registry = whenRegistryConfiguredWith(hook);
+            thenNoMetersRegistered(registry);
+        }
     }
 
     @Test
     public void testEmptyBinderNames() {
-        StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(List.of()));
-        MeterRegistry registry = whenRegistryConfiguredWith(hook);
-        thenNoMetersRegistered(registry);
+        try (StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(List.of()))) {
+            MeterRegistry registry = whenRegistryConfiguredWith(hook);
+            thenNoMetersRegistered(registry);
+        }
     }
 
     @Test
     public void testKnownBinder() {
-        StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(List.of("UptimeMetrics")));
-        MeterRegistry registry = whenRegistryConfiguredWith(hook);
-        thenUptimeMeterRegistered(registry);
+        try (StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(List.of("UptimeMetrics")))) {
+            MeterRegistry registry = whenRegistryConfiguredWith(hook);
+            thenUptimeMeterRegistered(registry);
+        }
     }
 
     @Test
     public void testUnknownBinder() {
-        StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(List.of("SadClown")));
-        assertThatThrownBy(() -> whenRegistryConfiguredWith(hook)).isInstanceOf(IllegalArgumentException.class);
+        try (StandardBindersHook hook = new StandardBindersHook(new StandardBindersHook.StandardBindersHookConfig(List.of("SadClown")))) {
+            assertThatThrownBy(() -> whenRegistryConfiguredWith(hook)).isInstanceOf(IllegalArgumentException.class);
+        }
     }
 
```

Comment:
Testing that close actually calls #close on the binder seems tricky.   There's no clean way to actually makes the observation.

Reply:
I found a way I could live with.